apps/cacvote-jx-terminal/backend/src/db.rs

```diff
@@ -68,18 +68,197 @@ pub(crate) async fn get_elections(
             }
         };
 
-        if let types_rs::cacvote::Payload::Election(election) = payload {
-            elections.push(election);
+        if let cacvote::Payload::Election(election) = payload {
+            elections.push(cacvote::ElectionPresenter::new(object.id, election));
         }
     }
 
     Ok(elections)
 }
 
+#[tracing::instrument(skip(connection))]
+pub async fn get_pending_registration_requests(
+    connection: &mut sqlx::PgConnection,
+) -> color_eyre::eyre::Result<Vec<cacvote::RegistrationRequestPresenter>> {
+    let records = sqlx::query!(
+        r#"
+        SELECT
+            rr.id,
+            rr.payload,
+            rr.certificates,
+            rr.signature,
+            rr.created_at
+        FROM
+            objects AS rr
+        WHERE
+            rr.object_type = $1
+          AND
+            NOT EXISTS (
+                SELECT 1
+                FROM objects AS r
+                WHERE r.object_type = $2
+                  AND rr.id = (convert_from(r.payload, 'UTF8')::jsonb ->> $3)::uuid
+            )
+        ORDER BY rr.created_at DESC
```

Comment:
@benadida mind taking a look at my poor man's foreign key join?

Reply:
that should work. Not super efficient because can't index the payload fields, but should be good for now.